src/videoConsumption.js

```diff
@@ -0,0 +1,155 @@
+    };
+
+    removeListener(listener) {
+        stopMeasurement();
+        super.removeListener(listener);
+    }
+}
+
+/**
+ * @type {Events.EventSingleton<videoDataCallback, VideoDataOptions>}
+ */
+export const onVideoData = new VideoDataEvent();
+
+/**
+ * A RegExp for the page match patterns.
+ * @type {RegExp|null}
+ */
+let matchPatternsRegExp = null;
+
+/**
+ * The registered page navigation content script.
+ * @type {RegisteredContentScript|null} 
+ */
+let registeredContentScript = null;
+
+/**
+ * Whether to notify the page data listener about private windows.
+ */
+let notifyAboutPrivateWindows = false;
+
+/**
+ * A function that is called when the content script sends a video data event message.
+ * @param {VideoDataDetails} videoData - Information about the video consumption.
+ */
+function videoDataListener(videoData) {
+    // If the page is in a private window and the module should not measure private windows,
+    // ignore the page
+    if(!notifyAboutPrivateWindows && videoData.privateWindow)
+        return;
+
+    // Delete the type string from the content script message
+    // There isn't (yet) a good way to document this in JSDoc, because there isn't support
+    // for object inheritance
+    delete videoData.type;
+
+    onVideoData.notifyListeners([ videoData ]);
+}
+
+/**
+ * Start a video consumption measurement. Note that only one measurement is currently supported per extension.
+ * @param {VideoDataOptions} options - A set of options for the measurement.
+ */
+export async function startMeasurement({
+    matchPatterns // causes error when empty, so we need to change default value
+}) {
+    let privateWindows = false
+
+    await pageManager.initialize();
+
+    matchPatternsRegExp = matching.matchPatternsToRegExp(matchPatterns);
+
+    notifyAboutPrivateWindows = privateWindows;
+
+    registeredContentScript = await browser.contentScripts.register({
+        matches: matchPatterns,
+        js: [{
+            file: inline.dataUrlToString(linkExposureContentScript)
+        }],
+        runAt: "document_idle"
+    });
+
+
+    // Listen for linkExposure messages from content script
+    messaging.onMessage.addListener(videoDataListener, {
+        type: "webScience.videoConsumption.videoData",
+        schema: {
+	        pageId: "string",
+	        url: "string",
+	        referrer: "string",
+	        pageVisitStartTime: "number",
+	        pageVisitStopTime: "number",
+	        playbackDuration: "number",
+	        ytChannel: "string", // TODO: what to put here if this might be empty? 
+	        privateWindow: "boolean"
+	    }
+    });
+}
+
+/**
+ * Stop a navigation measurement.
+ */
+function stopMeasurement() {
+    messaging.unregisterListener("WebScience.Measurements.VideoConsumption.VideoData", videoDataListener)
+    registeredContentScript.unregister();
+    registeredContentScript = null;
+    notifyAboutPrivateWindows = false;
+    matchPatternsRegExp = null;
+}
```

Comment:
See the `pageNavigation` and `linkExposure` PRs for examples of how to do per-listener teardown.